compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -783,6 +796,50 @@ impl<'ast, 'ra: 'ast, 'tcx> LateResolutionVisitor<'_, 'ast, 'ra, 'tcx> {
         (false, candidates)
     }
 
+    fn lookup_doc_alias_name(&mut self, path: &[Segment], ns: Namespace) -> Option<DefId> {
+        let item_str = path.last().unwrap().ident;
+
+        let find_doc_alias_name = |r: &mut Resolver<'ra, '_>, m: Module<'ra>| {
+            for resolution in r.resolutions(m).borrow().values() {
+                let Some(did) =
+                    resolution.borrow().binding.and_then(|binding| binding.res().opt_def_id())
+                else {
+                    continue;
+                };
+                if did.is_local() {
+                    // We don't record the doc alias name in the local crate
+                    // because the people who write doc alias are usually not
+                    // confused by them.
+                    continue;
+                }
+                let symbols = collect_doc_alias_symbol_from_attrs(r.tcx.get_attrs(did, sym::doc));
+                if symbols.contains(&item_str.name) {
+                    return Some(did);
+                }
+            }
+            None
+        };
+
+        if path.len() == 1 {
+            for rib in self.ribs[ns].iter().rev() {
+                if let RibKind::Module(module) = rib.kind
+                    && let Some(did) = find_doc_alias_name(self.r, module)
+                {
+                    return Some(did);
+                }
+            }
+        } else {
+            let mod_path = &path[..path.len() - 1];
+            if let PathResult::Module(ModuleOrUniformRoot::Module(module)) =
+                self.resolve_path(mod_path, Some(TypeNS), None)
```

Comment:
I don't think the logic is right, we are reporting an error for *one* specific unresolved segment in a path (and it's not necessarily the last one), not for all segments, and all segments before the one unresolved segment should be already resolved and recorded into `partial_res_map`.

So
- no need to search candidates for all segments, only for the currently unresolved one
- no need to use `resolve_path` for the previous segments, `partial_res_map` should already be filled.